pkg/sdk/shares.go

```diff
@@ -0,0 +1,311 @@
+}
+
+type ShareKind string
+
+const (
+	ShareKindInbound  ShareKind = "INBOUND"
+	ShareKindOutbound ShareKind = "OUTBOUND"
+)
+
+type Share struct {
+	CreatedOn    time.Time
+	Kind         ShareKind
+	Name         ExternalObjectIdentifier
+	DatabaseName AccountObjectIdentifier
+	To           []AccountIdentifier
+	Owner        string
+	Comment      string
+}
+
+func (v *Share) ID() AccountObjectIdentifier {
+	return v.Name.objectIdentifier.(AccountObjectIdentifier)
+}
+
+func (v *Share) ExternalID() ExternalObjectIdentifier {
+	return v.Name
+}
+
+type shareRow struct {
+	CreatedOn    time.Time `db:"created_on"`
+	Kind         string    `db:"kind"`
+	Name         string    `db:"name"`
+	DatabaseName string    `db:"database_name"`
+	To           string    `db:"to"`
+	Owner        string    `db:"owner"`
+	Comment      string    `db:"comment"`
+}
+
+func (r *shareRow) toShare() *Share {
+	toAccounts := strings.Split(r.To, ",")
+	var to []AccountIdentifier
+	if len(toAccounts) != 0 {
+		for _, a := range toAccounts {
+			if a == "" {
+				continue
+			}
+			parts := strings.Split(a, ".")
+			if len(parts) == 1 {
+				accountLocator := parts[0]
+				to = append(to, NewAccountIdentifierFromAccountLocator(accountLocator))
+				continue
+			}
+			orgName := parts[0]
+			accountName := strings.Join(parts[1:], ".")
+			to = append(to, NewAccountIdentifier(orgName, accountName))
+		}
+	}
+	return &Share{
+		CreatedOn:    r.CreatedOn,
+		Kind:         ShareKind(r.Kind),
+		Name:         NewExternalObjectIdentifierFromFullyQualifiedName(r.Name),
+		DatabaseName: NewAccountObjectIdentifier(r.DatabaseName),
+		To:           to,
+		Owner:        r.Owner,
+		Comment:      r.Comment,
+	}
+}
+
+type ShareCreateOptions struct {
+	create    bool                    `ddl:"static" db:"CREATE"` //lint:ignore U1000 This is used in the ddl tag
+	OrReplace *bool                   `ddl:"keyword" db:"OR REPLACE"`
+	share     bool                    `ddl:"static" db:"SHARE"` //lint:ignore U1000 This is used in the ddl tag
+	name      AccountObjectIdentifier `ddl:"identifier"`
+	Comment   *string                 `ddl:"parameter,single_quotes" db:"COMMENT"`
+}
+
+func (opts *ShareCreateOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return fmt.Errorf("not a valid object identifier: %s", opts.name)
+	}
+	return nil
+}
+
+func (v *shares) Create(ctx context.Context, id AccountObjectIdentifier, opts *ShareCreateOptions) error {
+	if opts == nil {
+		opts = &ShareCreateOptions{}
+	}
+	opts.name = id
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	clauses, err := v.builder.parseStruct(opts)
+	if err != nil {
+		return err
+	}
+	sql := v.builder.sql(clauses...)
+	_, err = v.client.exec(ctx, sql)
+	return decodeDriverError(err)
+}
+
+func (v *shares) Drop(ctx context.Context, id AccountObjectIdentifier) error {
+	sql := fmt.Sprintf(`DROP SHARE %s`, id.FullyQualifiedName())
+	_, err := v.client.exec(ctx, sql)
+	return decodeDriverError(err)
+}
```

Comment:
That makes it impossible to test the generated sql, I think it's worth sticking to using structs to describe the sql, even for these simple cases.